### PR TITLE
CVE-2017-7675

### PR DESCRIPTION
**Pull request by rmnill202**
Summary
===
- Added basic details for CVE-2017-7675

### CODE REVIEW

**Review by sfm2686 — COMMENTED** [review]
I could not comment no that specific line so I will make a general comment.

Do you think `defense_in_depth` applies here? If there were layers for protection and validation wouldn't the traversal outside of the module would be more difficult?

**Comment by sfm2686** [comment]
+1 upvote

File: cves/CVE-2017-7675.yml [hunk 1]
```diff
@@ -173,8 +203,10 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: |
```
Comment (s-newman) [inline cves/CVE-2017-7675.yml:213]: Consider revising to "...input not being properly parsed..." for smoother reading.

File: cves/CVE-2017-7675.yml [hunk 2]
```diff
@@ -204,4 +236,13 @@ mistakes:
...
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    Based on the comments surrounding the original implementation, as well as
+    the brief history of commits revolving around the chunk of code containing
+    the vulnerability and the lack of targeted unit tests; this vulnerability
+    was most certainly caused by a lack of proper risk assessment. 
+
+    With a comment like "This is almost certainly wrong and needs to be decoded",
```
Comment (s-newman) [inline cves/CVE-2017-7675.yml:251]: Was that comment actually in the code or commit history? Was there any explanation provided for why it was decided that the code should be accepted?

File: cves/CVE-2017-7675.yml [hunk 3]
```diff
@@ -4,33 +4,33 @@ nickname_instructions: |
   Nickname is optional. Provide a useful, professional, and catchy nickname for
   this vulnerability. Ideally fewer than 30 characters. This will be shown
   alongside its CVE to make it more easily distinguished from the rest.
-nickname:
+nickname: Tomcat URL Traverrsal
```
Comment (s-newman) [inline cves/CVE-2017-7675.yml]: Typo in `Traverrsal`, should be `Traversal`

**Comment by s-newman** [comment]
+4 upvotes. This is pretty neat!

File: cves/CVE-2017-7675.yml (hunk 3 above)
Comment (rmnill202) [reply cves/CVE-2017-7675.yml]: Good catch! :+1:

File: cves/CVE-2017-7675.yml (hunk 1 above)
Comment (rmnill202) [reply cves/CVE-2017-7675.yml:213]: Will do!

File: cves/CVE-2017-7675.yml (hunk 2 above)
Comment (rmnill202) [reply cves/CVE-2017-7675.yml:251]: Will clarify! The vulnerable code ended up replacing some code with the comment "TODO: This is almost certainly wrong and needs to be decoded".

**Comment by rmnill202** [comment]
> I could not comment no that specific line so I will make a general comment.
> 
> Do you think `defense_in_depth` applies here? If there were layers for protection and validation wouldn't the traversal outside of the module would be more difficult?

That sounds like a pretty good idea! Will update this field.

**Pull request merged**